src/Oauth2Client.php

```diff
@@ -0,0 +1,358 @@
+  }
+
+
+  /**
+   * Get information for the provided orderId
+   *
+   * @link https://developers.shapeways.com/api-reference#Order
+   *
+   * @param int $oderId the orderId for the Order to retreive
+   * @return array the json response from the api call
+   */
+  public function getOrderInfo($oderId)
+  {
+    $url = $this->baseApiUrl . '/orders/' . $oderId . '/' . $this->apiVersion;
+    return $this->_get($url);
+  }
+
+  /**
+   * @param $url
+   * @param array $params
+   * @param array $headers
+   * @param array $auth
+   * @return mixed
+   */
+  private function _post($url, $params = array(), $headers = array(), $auth = array()) {
+    $client = new \GuzzleHttp\Client();
+    if (array_key_exists('Content-type', $headers) && $headers['Content-type'] == 'application/json') {
+      $postOptions = array(\GuzzleHttp\RequestOptions::JSON => $params);
+    } else {
+      $postOptions = array(\GuzzleHttp\RequestOptions::FORM_PARAMS => $params);
+    }
+
+    if (!empty($headers)) {
+      $postOptions['headers'] =  $headers;
+    }
+
+    if (!empty($auth)) {
+      $postOptions['auth'] = $auth;
+    }
+
+    $res = $client->request('post', $url, $postOptions);
+
+    //echo $res->getStatusCode(); // "200"
+    //echo $res->getHeader('content-type'); // 'application/json; charset=utf8'
+    $response =  (string) $res->getBody(); // {"type":"User"...'
+    return json_decode($response);
+  }
+
+  /**
+   * @param $url
+   * @return mixed
+   */
+  private function _get($url)
+  {
+    $client = new \GuzzleHttp\Client();
+    try {
+      $res = $client->request('get', $url, array(
+        'headers' => array(
+          'Authorization' => 'Bearer ' . $this->accessToken,
+          'Content-type' =>  'application/json'
+        )
+      ));
+    } catch (\Exception $e) {
+      echo $e->getMessage();
+    }
+    //echo $res->getStatusCode(); // "200"
+    //echo $res->getHeader('content-type'); // 'application/json; charset=utf8'
+    $response = (string)$res->getBody(); // {"type":"User"...'
+    return json_decode($response);
+  }
+
+  /**
+   * Make a POST request to the api server
+   *
+   * @param string $url the api url to request
+   * @param array $params the parameters to send with the request
+   * @param array $headers to send with the request
+   * @param array $auth to send with the request
+   * @return array the json response from the api call
+   */
+  private function _postCurl($url, $params = array(), $headers= array(), $auth = array())
+  {
+    // json encode
+    $postData = json_encode($params);
+
+    $ch = curl_init($url);
+
+    if (!empty($headers)) {
+      $headers = array();
+      foreach ($headers as $optKey => $optVal) {
+        $headers[] = $optKey.': '.$optVal;
+      }
+      curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+    }
+
+    if (!empty($auth)) {
+      curl_setopt($ch, CURLOPT_USERPWD, $auth[0] . ":" . $auth[1]);
+    }
+
+    curl_setopt($ch, CURLOPT_TIMEOUT, 30);
```

Comment:
should we keep using curl or inject Guzzle library? I think for simplicity curl is fine

Reply:
curl functions are not used